spring-cloud-dataflow-completion/src/main/java/org/springframework/cloud/dataflow/completion/StreamCompletionProvider.java

```diff
@@ -35,6 +39,15 @@
 	@Autowired
 	private List<? extends ExpansionStrategy> completionExpansionStrategies = new ArrayList<>();
 
+	@PostConstruct
+	public void sortStrategies() {
+		// Here we mean HIGHEST_PRECEDENCE to win over others, so must be *last*
+		AnnotationAwareOrderComparator.sort(completionExpansionStrategies);
```

Comment:
This seems like an abuse of the `@Ordered` annotation to avoid the real problem which is a result of using field injection for a collection. We need to remove all field injection eventually, so I'd rather see this change to constructor injection and explicit configuration of the desired order within the list.